FavoriteArtists/MKJAddArtistViewController.m

```diff
@@ -0,0 +1,65 @@
+//
+//  MKJAddArtistViewController.m
+//  FavoriteArtists
+//
+//  Created by Kenneth Jones on 12/3/20.
+//
+
+#import "MKJAddArtistViewController.h"
+
+@interface MKJAddArtistViewController ()
+
+@property (weak, nonatomic) IBOutlet UISearchBar *artistSearchBar;
+@property (weak, nonatomic) IBOutlet UILabel *artistLabel;
+@property (weak, nonatomic) IBOutlet UILabel *yearLabel;
+@property (weak, nonatomic) IBOutlet UILabel *biographyLabel;
+
+@property (nonatomic) NSString *artistName;
+@property (nonatomic) int year;
+@property (nonatomic) NSString *biography;
+
+@end
+
+@implementation MKJAddArtistViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    _artistSearchBar.delegate = self;
+    _artistLabel.text = @"";
+    _yearLabel.text = @"";
+    _biographyLabel.text = @"";
+}
+
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar {
+    [self.fetcher fetchArtistWithArtist:searchBar.text completionHandler:^(MKJArtist *artist, NSError *error) {
+        self.artistLabel.text = artist.artist;
+        self.artistName = artist.artist;
+        self.yearLabel.text = [NSString stringWithFormat:@"Formed in %i", artist.year];
+        self.year = artist.year;
+        self.biographyLabel.text = artist.biography;
+        self.biography = artist.biography;
+    }];
+}
+
+- (IBAction)saveArtist:(id)sender {
+    if ([self.artistLabel.text isEqualToString:@""]) {
+        [self.navigationController popViewControllerAnimated:true];
+    } else {
+        MKJArtist *artistInfo = [[MKJArtist alloc] initWithArtist:_artistName year:_year biography:_biography];
+        [self.fetcher saveArtist:artistInfo];
+        [self.navigationController popViewControllerAnimated:true];
+    }
+}
```

Comment:
I really like that you are checking to see if the there is an artist before trying to save. It looks like you ran out of time on this sprint challenge, but maybe adding an alert to the user and not popping back to the previous vc might make more sense here. Also one other suggestion, I would maybe try to create the artist object in the search bar method  and then checking if the artist object is nil or not in the saveArtist action. I don't think its applicable here but say there was an artist who didn't have that specific property on it. I just think its a little bit safer way to check.